prow/crier/controller.go

```diff
@@ -218,24 +218,9 @@ func (c *Controller) processNextItem() bool {
 		return true
 	}
 
+	// we set omitempty on PrevReportStates, so here we need to init it if is nil
 	if pj.Status.PrevReportStates == nil {
-		// old prowjobs doesn't have this field, let's update it with an empty map...
-
-		pjCopy := pj.DeepCopy()
-		pjCopy.Status.PrevReportStates = map[string]v1.ProwJobState{}
-		if _, err := c.pjclientset.Prow().ProwJobs(pjCopy.Namespace).Update(pjCopy); err != nil {
-			logrus.WithError(err).WithField("prowjob", keyRaw).Error("failed to update pj")
-			return c.retry(key, err)
-		}
-
-		//resync with apiserver
-		pj, err = c.pjclientset.Prow().ProwJobs(pj.Namespace).Get(pj.Name, metav1.GetOptions{})
-		if err != nil {
-			// hope next cycle will catch up
-			logrus.WithError(err).WithField("prowjob", keyRaw).Error("failed to resync pj")
-			c.queue.Forget(key)
-			return true
-		}
+		pj.Status.PrevReportStates = map[string]v1.ProwJobState{}
```

Comment:
this is from https://github.com/kubernetes/test-infra/pull/9747, it will always be empty first time in crier, and above logic won't work anymore :-\

Reply:
cc @BenTheElder

Reply:
cc @stevekuznetsov depending on some other package to setup your pointers is a recipe for outages :)

Reply:
@fejta :joy so actually everything happens in the same package: I was creating the map and trying to resync with apiserver but it won't take the empty map anymore :man_shrugging: